pkg/api/vlabs/types.go

```diff
@@ -277,7 +277,7 @@ type KubernetesConfig struct {
 	UserAssignedID                  string            `json:"userAssignedID,omitempty"`
 	UserAssignedClientID            string            `json:"userAssignedClientID,omitempty"` //Note: cannot be provided in config. Used *only* for transferring this to azure.json.
 	CustomHyperkubeImage            string            `json:"customHyperkubeImage,omitempty"`
-	DockerEngineVersion             string            `json:"dockerEngineVersion,omitempty"`
+	DockerEngineVersion             string            `json:"dockerEngineVersion,omitempty"` // Deprecated
```

Comment:
we might want to add some sort of validation/warning message that DockerEngineVersion is deprecated and Docker is replaced by Moby for customers still passing in a Docker engine version. Thinking of customers with working apimodels specifying a Docker Engine Version that always use the same apimodels.

Reply:
good idea, done